taichi/transforms/vector_split.cpp

```diff
@@ -241,8 +242,11 @@ class BasicBlockVectorSplit : public IRVisitor {
 
   void visit(PrintStmt *stmt) override {
     for (int i = 0; i < current_split_factor; i++) {
-      current_split[i] =
-          Stmt::make<PrintStmt>(lookup(stmt->stmt, i), stmt->str);
+      std::vector<Stmt *> new_contents;
+      std::transform(stmt->contents.begin(), stmt->contents.end(),
+                     new_contents.begin(),
+                     [&](auto &&x) { return lookup(x, i); });
```

Comment:
FYI: Note that this is actually capturing `i` by **reference**, which could be dangerous sometimes. Fortunately, `std::transform` finishes synchronously. Otherwise, the `i` in `return lookup(x, i);` could be different from when the lambda was constructed. I don't think this is a problem here, but I would prefer `[=]` just to be safe.

Reply:
Sorry let me just add a few more comments on this: I'm suggesting to use `[=]` because this is inside a loop. If you used Go before, you probably know that there's a nasty bug, where if you create an anonymous function that captures the index `i` of the loop, it is captured by reference. Since lots of the times these anonymous functions are goroutines, they run asynchronously, and the index value would be changed inside the function. See https://yourbasic.org/golang/gotcha-data-race-closure/